python/lsst/ip/isr/isrFunctions.py

```diff
@@ -216,6 +217,108 @@ def growMasks(mask, radius=0, maskNameList=['BAD'], maskValue="BAD"):
         fpSet.setMask(mask, maskValue)
 
 
+def maskITLEdgeBleed(ccdExposure, itlEdgeBleedSatMinArea,
+                     itlEdgeBleedSatFracLevel, itlEdgeBleedModelConstant, saturatedMaskName, badAmpDict):
+    """Mask edge bleeds in ITL detectors.
+
+    Parameters
+    ----------
+    ccdExposure : `lsst.afw.image.Exposure`
+        Exposure to apply masking to.
+    itlEdgeBleedSatMinArea : scalar
+        Minimal saturated footprint area where the presence of edge bleeds
+        will be checked.
+    itlEdgeBleedSatFracLevel : scalar
+       Fraction of the saturation level at the detector edge
+       above which there is an edge bleed.
+    itlEdgeBleedModelConstant : scalar
+        Constant in the decaying exponential in the edge bleed masking.
+    saturatedMaskName : `str`
+        Mask name for saturation.
+    badAmpDict : `str`[`bool`]
+        Dictionary of amplifiers, keyed by name, value is True if
+        amplifier is fully masked.
+    """
+    maskedImage = ccdExposure.maskedImage
+
+    # Get minimum of amplifier saturation level
+    satLevel = numpy.min([ccdExposure.metadata[f"LSST ISR SATURATION LEVEL {amp.getName()}"]
+                          for amp in ccdExposure.getDetector() if not badAmpDict[amp.getName()]])
+
+    # Get footprint list of saturated pixels
+    thresh = afwDetection.Threshold(satLevel)
+    fpList = afwDetection.FootprintSet(maskedImage, thresh).getFootprints()
+
+    satAreas = numpy.asarray([fp.getArea() for fp in fpList])
+    largeAreas, = numpy.where(satAreas >= itlEdgeBleedSatMinArea)
+
+    if len(largeAreas) == 0:
+        return
+    else:
+        for largeAreasIndex in largeAreas:
+
+            # Get centroid of saturated core
+            xCore, yCore = fpList[largeAreasIndex].getCentroid()
+            # Turn the Y detector coordinate
+            # into Y footprint coordinate
+            yCoreFP = yCore - fpList[largeAreasIndex].getBBox().getMin().getY()
+            # Get the number of saturated columns around the centroid
+            widthSat = numpy.sum(fpList[largeAreasIndex].getSpans().asArray()[int(yCoreFP), :])
+
+            for amp in ccdExposure.getDetector():
+                # Select 2 amplifiers around the saturated
+                # core with a potential edge bleed
+                yBox = amp.getBBox().getCenter()[1]
+                if amp.getBBox().contains(xCore, yBox):
```

Comment:
I'm missing how this picks up two amplifiers.

Reply:
It will pick up the 2 amplifiers that have the same x coordinate. We don't care about the y coordinate so we put the center of the amp bbox for y to simplify.

Reply:
Oh I see.  Can you explicitly add this to the comment?  Thanks!